### ISSUE TITLE
[WordPress 5.5 beta] CPT with locked template: Adding soft line breaks in a paragraph block is not working anymore

### ISSUE THREAD

**Issue by them-es** [WordPress/gutenberg#24124]
**Describe the bug**
Adding soft line breaks in core/paragraph blocks of *locked* Custom Post type templates using the "template_lock" property is not working anymore.

**To reproduce**
1. Register a custom post type with a template and include **'template_lock' => 'all'**
```
function myplugin_register_book_post_type() {
    $args = array(
        'public' => true,
        'label'  => 'Books',
        'show_in_rest' => true,
        'template_lock' => 'all',
        'template' => array(
            array( 'core/image', array(
                'align' => 'left',
            ) ),
            array( 'core/heading', array(
                'placeholder' => 'Add Author...',
            ) ),
            array( 'core/paragraph', array(
                'placeholder' => 'Add Description...',
            ) ),
        ),
    );
    register_post_type( 'book', $args );
}
add_action( 'init', 'myplugin_register_book_post_type' );
```
2. Create a new book post.
3. Try to add a line break in the description block using **Shift+Enter** and notice that a whitespace character appears instead of a new line.
4. Without the 'template_lock' property everything works as expected.

**Expected behavior**
Adding line breaks to paragraphs should also be possible on *locked* templates. Like in WordPress <=5.4!

**Editor version (please complete the following information):**
- WordPress version: 5.5 beta3
- Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? "default"

**Desktop (please complete the following information):**
 - OS: macOS
 - Browser: Chrome
 - Version: 84

**Comment by ocean90** [comment]
Confirmed and can be reproduced with the following code for the browser console:


```js
( function( blocks, element, blockEditor ) {
    var el = element.createElement;
    var InnerBlocks = blockEditor.InnerBlocks;
 
    blocks.registerBlockType( 'gutenberg-tests/template-lock', {
        title: 'Template Lock Test',
        category: 'common',
 
        edit: function( props ) {
            return el(
                'div',
                { className: props.className },
                el(
                    InnerBlocks,
                    {
                        template: [ [ 'core/paragraph', { placeholder: 'Enter something…' } ] ],
                        templateLock: "all",
                    }
                )
            );
        },
 
        save: function( props ) {
            return el(
                'div',
                { className: props.className },
                el( InnerBlocks.Content )
            );
        },
    } );
} (
    window.wp.blocks,
    window.wp.element,
    window.wp.blockEditor,
) );
```

It happens also with `templateLock` set to `"insert"`.  Shift+Enter no longer adds a line break.

**Comment by ocean90** [comment]
Seems to be caused by this line:
https://github.com/WordPress/gutenberg/blob/b9de73ddb58b6b88faa4846d3f8294c815c9d4df/packages/block-editor/src/components/block-list/block-wrapper.js#L261-L262

This was introduced in https://github.com/WordPress/gutenberg/commit/67e2df7275de5972f9aca83bceabbed6ae396b81 and doesn't seem to be intentionally. Can you confirm this, @ellatrix?

**Comment by ellatrix** [comment]
I'll look into it

**Comment by ellatrix** [comment]
What I'm seeing if I run the above code in the console is `Enter` not working if the template is locked (cannot reproduce a space being inserted). This makes sense because the normal behaviour is splitting by inserting another paragraph, which is disabled. For consistency, only `Shift+Enter` inserts a soft line break.

**Comment by ocean90** [comment]
>  For consistency, only Shift+Enter inserts a soft line break.

Correct, and that's what's broken.

**Comment by ocean90** [comment]
Just found a duplicate report with an open PR for this: https://github.com/WordPress/gutenberg/pull/23330

**Comment by ellatrix** [comment]
Oh, sorry for misreading the problem.

**Comment by ocean90** [comment]
Fixed by #23330.
